mesonbuild/linkers/linkers.py

```diff
@@ -705,7 +705,9 @@ def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
         args.extend(self._apply_prefix('-rpath,' + paths))
 
         # TODO: should this actually be "for solaris/sunos"?
-        if mesonlib.is_sunos():
+        # NOTE: Remove the zigcc check once zig support "-rpath-link"
+        # See https://github.com/ziglang/zig/issues/18713
```

Comment:
Can we grep the help output from zig to know if it supports this?

Reply:
Or we can do a version check

Reply:
Version check yes, help check would probably be slower and might display the wrong thing (not 100% sure).

Reply:
Hmm I guess support for this flag hasn't been planned, so the id check makes sense for now.

Reply:
Yeah hence the comment with the link. We can change it once the flag exists in a release version.